### ISSUE TITLE
Upgrade yamerl to 0.10.0

### ISSUE THREAD

**Issue by Sgoettschkes** [launchdarkly/erlang-server-sdk#77]
**Describe the bug**
The yamerl library version 0.10 was released in January of 2022 and most libraries have upgraded by now, with some now requiring yamerl 0.10.0 as the minimum version. For example, the newer versions of the elixir library mix_audit depend on yamerl 0.10 and we are not able to upgrade since the launchdarkly sdk still uses 0.8.1

**To reproduce**
Create a new elixir project using mix, add

```
{:ldclient, "~> 2.0.0", hex: :launchdarkly_server_sdk},
{:mix_audit, "~> 2.1", only: [:dev, :test], runtime: false}
```

as dependencies and run `mix deps.get`. Result:

```
Because mix_audit >= 2.1.0 depends on yaml_elixir ~> 2.9 which depends on yamerl ~> 0.10, mix_audit >= 2.1.0 requires yamerl ~> 0.10.
And because launchdarkly_server_sdk >= 1.2.0 depends on yamerl 0.8.1, mix_audit >= 2.1.0 is incompatible with launchdarkly_server_sdk >= 1.2.0.
And because your app depends on launchdarkly_server_sdk ~> 2.0.0, mix_audit >= 2.1.0 is forbidden.
So, because your app depends on mix_audit ~> 2.1, version solving failed.
```

**SDK version**
2.0.04

**Language version, developer tools**
Elixir 14.2, Erlang 25

**Comment by kinyoklion** [comment]
Hello @Sgoettschkes,

Thank you for pointing this out. Will look into it, seems like it should be a straightforward update.

_Filed internally as 206285_

Thank you,
Ryan

**Comment by kinyoklion** [comment]
Hello @Sgoettschkes,

This has been released in 2.0.5.

Thank you,
Ryan

**Comment by Sgoettschkes** [comment]
I did the update and can confirm that yamerl has been updated and other dependencies are also up-to-date.

Thank you very much for the fast response!
